microsoft/testsuites/core/temp_vm_hot_resize.ps1

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Description
+    It deploys the VM, verifies core number/memory size,
+    then resize the VM, verifies new core number/memory size.
+#>
+
+param([object] $AllVMData)
```

Comment:
What method would pass in these necessary params in LISAv3

Reply:
Hey @squirrelsc do you have any insignts on this question?

Reply:
You can refer to current feature to implement it as a feature. In my mind (may be wrong), there are several methods in the feature.
1. check if current size is resizable. If not, skip this test case.
2. get list of resizable candidates.
3. pick one from it (maybe smaller one), and then resize, verify.

The feature may be like,

```python
class Resize(Feature):
    def is_supported(self) -> bool:
        ...

    def resize(self) -> None:
        ...
```

Reply:
How can I get the info found in $AllVMData?

Reply:
try to get vm_size and other fields by `node.capability.get_extended_runbook(AzureNodeSchema, AZURE)`. If you search `AzureNodeSchema`, you can find its structure and examples in azure folder.